### ISSUE TITLE
The index directive with "builtin" type generates different index entry than the function directive

### ISSUE THREAD

**Issue by serhiy-storchaka** [sphinx-doc/sphinx#7219]
```
.. index::
   builtin: max
```
and
```
.. function:: max(iterable, *[, key, default])
```
generate similar but different index entries.

The former generates:
```
max
    built-in function
```
The latter generates:
```
max() (built-in function)
```
Differences:

1. Parenthesis are added or not after the name of the function.
2. Parenthesis are added or not around "built-in function".
3. It takes one or two lines.
4. "max" is a link or not.

Link: https://docs.python.org/3/genindex-M.html

This issue was reported on the Python bug tracker: https://bugs.python.org/issue21352.

Would be nice to unify and merge index entries generated by "function" and "index".

**Comment by tk0miya** [comment]
Unfortunately, python specific index types (like `builtin`) were deprecated in 10 years ago: https://github.com/sphinx-doc/sphinx/commit/3e9182550a#diff-238bd7cedf25808f7e89537fbcd798dcR188-R189
And we are considering to remove it in future: #6970 . But we need to consider about it more deeply...

**Comment by serhiy-storchaka** [comment]
What should be used in the Python documentation?

**Comment by tk0miya** [comment]
Now the deprecation of python specific index types is postponed (see #6970). So I just made a fix for this #7395.
